modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/config/ApplicationConfiguration.java

```diff
@@ -46,4 +52,26 @@ public PasswordEncoder passwordEncoder() {
 
         return new DelegatingPasswordEncoder(encodingId, encoders);
     }
+
+    /**
+     * @return Application event multicaster.
+     */
+    @Bean(name = "applicationEventMulticaster")
+    public ApplicationEventMulticaster simpleApplicationEventMulticaster(TaskExecutor executor) {
+        SimpleApplicationEventMulticaster evtMulticaster = new SimpleApplicationEventMulticaster();
+
+        evtMulticaster.setTaskExecutor(getThreadPoolTaskExecutor());
+        return evtMulticaster;
+    }
+
+    /**
+     * Thread pool task executor.
+     */
+    private Executor getThreadPoolTaskExecutor() {
+        ThreadPoolTaskExecutor pool = new ThreadPoolTaskExecutor();
+        pool.setMaxPoolSize(Runtime.getRuntime().availableProcessors());
```

Comment:
Look at org.apache.ignite.configuration.IgniteConfiguration#DFLT_PUBLIC_THREAD_CNT
I think we should limit upper bound

Reply:
Done.